modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/datastructure/DataStructure.java

```diff
@@ -183,6 +187,20 @@ protected final long allocatePage(@Nullable ReuseBag bag, boolean useRecycled) t
         return pageId;
     }
 
+    /**
+     * Replaces the "partition ID" part of the given page ID with the partition ID that his data structure is responsible for.
+     *
+     * @param pageId Original page ID.
+     * @return Page ID with replaced partition ID.
+     */
+    private long replacePartitionId(long pageId) {
+        long partitionIdZeroMask = ~(PageIdUtils.PART_ID_MASK << PageIdUtils.PAGE_IDX_SIZE);
+
+        long partitionIdMask = ((long) partId) << PageIdUtils.PAGE_IDX_SIZE;
```

Comment:
You may ignore this comment. What you store in this variable is not technically a mask, it's just a value

Reply:
yes, I know that, but `partitionIdPart` is worse and I didn't come up with a better name